unit_tests/src/test_server_properties.cpp

```diff
@@ -56,3 +56,27 @@ TEST_CASE("server_properties is threadsafe", "[thread-safety]")
         }
     }
 }
+
+TEST_CASE("server_properties map function is threadsafe", "[thread-safety]")
+{
+    const auto limit{1000};
+
+    // Start a thread that will generally get in the way a lot by writing
+    // to the irods::server_properties object
+    std::thread persistent_write([]() {
+        for (auto i = 0; i < limit; i++) {
+            irods::set_server_property(std::to_string(i), i);
+        }
+    });
+    irods::at_scope_exit([&persistent_write]() { persistent_write.join(); });
```

Comment:
Your `at_scope_exit` needs a name.

`persistent_write` joins before the logic below is ever executed.